bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java

```diff
@@ -667,7 +684,21 @@ private boolean updateRollerStatus(ShellySettingsStatus status, @Nullable Shelly
             getThing().postEvent(SHELLY_EVENT_ROLLER_CALIB, false);
         }
 
-        return updateChannels ? ShellyComponents.updateRoller((ShellyBaseHandler) getThing(), rs, cs.id) : false;
+        return updateChannels ? ShellyComponents.updateRoller((ShellyBaseHandler) getThing(), rs, rIdx) : false;
+    }
+
+    private int getRollerIdx(ShellyDeviceProfile profile, @Nullable Integer id) {
+        List<ShellySettingsRoller> rollers = profile.settings.rollers;
+        if (id != null && rollers != null) {
+            int idx = 0;
+            for (ShellySettingsRoller roller : rollers) {
+                if (roller.isValid && roller.id != null && roller.id.intValue() == id.intValue()) {
+                    return idx;
+                }
+                idx++;
+            }
+        }
+        return -1;
```

Comment:
Won't this cause `IndexOutOfBoundsException` in lines 630-632?

Reply:
no, this is checked by the caller and leads into IllegalArgumentException

Reply:
I don't understand your comment. It was not checked by the caller. But I see that you now throw `IllegalArgumentException` instead of returning -1, so `IndexOutOfBoundsException` should no longer happen.

Reply:
yes